rodeo/src/rodeo/06-either-and-try.scala

```diff
@@ -0,0 +1,186 @@
+package rodeo
+import zio.test._
+
+object EitherAndTry extends Chapter {
+
+  import scala.util.{Try, Success, Failure}
+
+  // The `Try[A]` type is similar to `Option[A]`. It is used to represent the type of a computation that attempts to return an instance of type A.
+  // If the computation succeeds we obtain an instance of type `Success[A]`. If it fails we obtain a wrapped instance of the Throwable class. This instance will have type `Failure`.
+  // The Throwable class is the least derived class of the Scala exception hierarchy. An instance of this class can represent an error or an exception.
+  // For practical purposes the main difference between `None` and `Failure` is that `Failure` is able to provide insight into the cause of the "failure".
+
+  Exercise("Try constructor") {
+    assertTrue(Try(7) == Success(???))
+
+    // Delete the following line if you can't find the answer.
+
+    assertTrue(Try(10 / 0).getClass == ???)
+  }
+
+  Exercise("Books") {
+    def bookRecomendation(
+        likesMath: Boolean,
+        likesComputerScience: Boolean
+    ): Try[String] =
+      if (!likesMath && !likesComputerScience)
+        Failure(
+          new IllegalArgumentException("You need to go visit a quantum doctor")
+        )
+      else if (likesMath && !likesComputerScience)
+        Failure(new IllegalArgumentException("You should not be reading this"))
+      else if (!likesMath && likesComputerScience)
+        Failure(
+          new IllegalArgumentException(
+            "I am unable to give you a recommendation"
+          )
+        )
+      else Success("The emperors new mind")
+
+    assertTrue(bookRecomendation(false, true) == ???)
+    assertTrue(bookRecomendation(true, true) == ???)
+  }
+
+// To find a solution of the quadratic equation ax^2+bx+c=0 we only need to use the formula  x = (-b + sqrt(b^2-4ac))/2a.
+
+  Exercise("Equation") {
+    def quadraticEquationSolver(
+        a: Double,
+        b: Double,
+        c: Double
+    ): Try[Double] = {
+      def discriminant(a: Double, b: Double, c: Double): Double =
+        Math.pow(b, 2) - 4 * a * c
+      if (discriminant(a, b, c) >= 0)
+        Success((-b + Math.sqrt(discriminant(a, b, c))) / 2 * a)
+      else if (a == 0)
+        Failure(
+          new IllegalArgumentException("This is not a quadratic equation")
+        )
+      else
+        Failure(
+          new IllegalArgumentException(
+            "This equations does not have a solution in the real numbers"
+          )
+        )
+    }
+
+    assertTrue(quadraticEquationSolver(0, 7, 13) == ???)
+    assertTrue(quadraticEquationSolver(???, ???, ???) == Success(Math.sqrt(2)))
```

Comment:
I think this exercise is more about the mathematical aspect, than testing out Try? I understand this as: find an equation which its discriminant is root of 2

I like this exercise, but I fear the assertions are more mathematical than programming?
Any thoughts?

Reply:
Looking at the `sqrt` excercise, perhaps we can implement that same one in terms of Try for this exercise